engine/class_modules/paladin/sc_paladin_protection.cpp

```diff
@@ -751,6 +771,20 @@ void paladin_t::trigger_holy_shield( action_state_t* s )
   active.holy_shield_damage -> schedule_execute();
 }
 
+void paladin_t::trigger_t28_4p_pp( action_state_t* s )
+{
+  // escape if we don't have t28 4p
```

Comment:
Please remove the comment lines here. Alternatively, you could move all of the conditions you have at the caller into this function.